### PR TITLE
feat: add new destination tiktok_audience

### PR DESCRIPTION
**Pull request by ItsSudip**
## Description of the change

> We are implementing a new audience type destination named Tiktok Audience.
Resolves INT-757

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/cdk/v2/destinations/tiktok_audience/procWorkflow.yaml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+          destinationFields.forEach(function(destinationField){
+            let hashedData = trait[destinationField] ? {
+              "id": isHashRequired ? $.SHA256(trait) : trait,
+              "audience_ids":[audienceId]
+            } :{};
+            arr.push(hashedData); 
+          })
+          batch_data.push(arr);
+        })
+        batch_data
```
Comment (sanpj2292) [inline src/cdk/v2/destinations/tiktok_audience/procWorkflow.yaml:43]: What does this line do ?

File: src/cdk/v2/destinations/tiktok_audience/procWorkflow.yaml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      };
```
Comment (sanpj2292) [inline src/cdk/v2/destinations/tiktok_audience/procWorkflow.yaml:44]:
```suggestion
      const hashTraits = function(traits) {
        const batch_data = traits.map(function(trait){
          const arr = destinationFields.map(function(destinationField){
            let hashedData = trait[destinationField] ? {
              "id": isHashRequired ? $.SHA256(trait) : trait,
              "audience_ids":[audienceId]
            } :{};
            return hashedData; 
          })
          return arr;
        })
        batch_data
      };
```
Can we refactor like this ?

**Comment by koladilip** [comment]
@ItsSudip I have refactored workflows to use features of json template engine, instead of writing functions all the time.

**Review by koladilip — APPROVED** [review]

**Review by koladilip — APPROVED** [review]

File: src/cdk/v2/destinations/tiktok_audience/rtWorkflow.yaml [hunk 3]
```diff
@@ -14,14 +14,16 @@ steps:
...
     template: |
-      $.outputs.transform#idx.output.({
-        "batchedRequest": .,
-        "batched": true,
-        "destination": ^[idx].destination,
-        "metadata": ^[idx].metadata,
-        "statusCode": 200
-      })[]
-
+      $.outputs.transform.map(function(transformedData,index){
```
Comment (koladilip) [inline src/cdk/v2/destinations/tiktok_audience/rtWorkflow.yaml:17]: any reason you are using map function, you can do this without it, using native json template engine features

**Review by koladilip — DISMISSED** [review]

**Review by koladilip — APPROVED** [review]

**Review by sanpj2292 — APPROVED** [review]

**Pull request merged**